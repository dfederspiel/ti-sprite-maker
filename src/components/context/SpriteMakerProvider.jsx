import React, { createContext, useContext, useState } from 'react';
import PropTypes from 'prop-types';
import { defaultGrid, getHex } from './constants';

export const useSpriteMakerProviderProps = () => {
  const [sprite] = useState('0'.repeat(16));

  return {
    sprite,
    getHex,
    defaultGrid,
  };
};

const spriteMakerContext = createContext(null);

export const SpriteMakerProvider = ({ children }) => {
  const props = useSpriteMakerProviderProps();
  return (
    <spriteMakerContext.Provider
      value={props}
    >
      {children}
    </spriteMakerContext.Provider>
  );
};

SpriteMakerProvider.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
    PropTypes.string,
  ]).isRequired,
};

export const useSpriteMaker = () => useContext(spriteMakerContext);
