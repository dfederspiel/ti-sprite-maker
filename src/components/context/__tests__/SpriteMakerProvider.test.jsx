import React from 'react';
import { render } from '@testing-library/react';
import { renderHook } from '@testing-library/react-hooks';
import { SpriteMakerProvider, useSpriteMaker } from '../SpriteMakerProvider';

describe('the sprite maker provider', () => {
  it('renders as expected', () => {
    const { asFragment } = render(
      <SpriteMakerProvider>TEST</SpriteMakerProvider>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});

describe('the sprite maker hook', () => {
  const wrapper = ({ children }) => (
    <SpriteMakerProvider>{children}</SpriteMakerProvider>
  );
  it('exposes the underlying context', () => {
    const { result } = renderHook(() => useSpriteMaker(), { wrapper });
    expect(result.current.sprite).toMatchSnapshot('and has a default sprite state');
  });
});
