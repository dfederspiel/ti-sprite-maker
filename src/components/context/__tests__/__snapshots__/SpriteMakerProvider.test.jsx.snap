// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the sprite maker hook exposes the underlying context: and has a default sprite state 1`] = `"0000000000000000"`;

exports[`the sprite maker provider renders as expected 1`] = `
<DocumentFragment>
  TEST
</DocumentFragment>
`;
